doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+dynamic profiles work?
+
+I propose we add functionality to automatically search for these profile sources
+and add/remove them on _every_ Terminal launch. To make this functionality work
+appropriately, we'll need to introduce a constraint on dynamic profiles.
+
+**For any dynamic profiles, they must be able to be generated using a stable
+GUID**. For example, any time we try adding the "Ubuntu" profile, we must be
+able to generate the same GUID every time. This way, when a dynamic profile
+generator runs, it can check if that profile source already has a profile
+associated with it, and do nothing (as to not create many duplicate "Ubuntu"
+profiles, for example).
+
+When a dynamic profile generator runs, it will determine what new profiles need
+to be added to the user settings, and it can append those to the list of
+profiles. The generator will return some sort of result indicating that it wants
+a save operation to occur. The app will then save the `profiles.json` file,
+including these new profiles.
+
+If a dynamic profile generator has determined that a profile should no longer be
+used, we don't want to totally remove the profile from the file - perhaps
+there's some data the user would like from those settings. Instead, we'll simply
+mark that profile as `"hidden": true`. That way, the profile will remain in
+their settings, but will not appear in the list of profiles.
+
+Should a dynamic profile generator try to create a profile that's already
+`hidden`, it'll make sure to set `hidden` back to false, making the original
+visible again.
+
+Additionally, a user might not want a dynamic profile generator to always run.
+They might want to keep their Azure connections visible in the list of profiles,
+even if its no longer a valid target. Or they might want to not automatically
+connect to Azure to find new instances every time they launch the terminal. To
+enable scenarios like this, **each dynamic profile generator must provide a way
+to disable it**. For the above listed cases, the two settings might be something
+like `autoloadWslProfiles` and `autoloadAzureConnections`. These will be set to
+true in the default settings, but the user can override them to false if they so
+chose.
+
+#### What if a dynamic profile is removed, but it's the default?
+
+We should probably add an extra validation step at the end of serializing to
+ensure that the default profile isn't a hidden profile. If the user's set
+default profile is a hidden profile, we'll just set the first non-hidden profile
+as the default profile.
+
+### Powershell Core & the Defaults
+
+How do we handle the potential existence of powershell core in this model?
+Powershell core is unique as far as the default profiles goes - it may or may
+not exist on the user's system. Not only that, but depending on the user's
+install of powershell core, it might have a path in either `Program Files` or
+`Program Files(x86)`.

```

Comment:
Easy. dynamic profile generator.

Reply:
This entire section actually could be "use a dynamic profile generator" 😸

Reply:
UGH NEVERMIND YOU BEAT ME TO IT
i should read ahead before i talk about "this entire section"